daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/DataProcessor.scala

```diff
@@ -340,7 +340,7 @@ class DataProcessor(
    * runtime. Instead we deal with success and failure statuses.
    */
   def parse(input: InputSourceDataInputStream, output: InfosetOutputter): DFDL.ParseResult = {
-    Assert.usage(!this.isError)
+    Assert.usage(!this.isError, this.getDiagnostics.mkString)
```

Comment:
Same comment as above, these message want to be about not calling parse/unparse when DataProcessor.isError is true.